evm-events-calls-raw/chain_configs.go

```diff
@@ -13,6 +13,9 @@ type ChainConfig struct {
 	DisplayName          string // Public
 	ExplorerLink         string
 	ApiEndpoint          string
+	ApiBaseURL           string // Base URL without query parameters
+	ApiQueryParams       string // Query parameters to append (e.g. "?chainid=747474")
+	ApiPathPattern       string // Path pattern for direct API calls (e.g. "/api", "/{address}")
```

Comment:
This seems to be tied to detect usage of `EtherscanV2API` vs Etherscan legacy API.

We should probably keep a single API URL and have instead an `api_type` field that can tell us what kind of API version we deal with.

Also for `ApiPathPattern`, I don't see a good usage, should be merged inside `ApiBaseURL` which should itself be equivalent to ApiEndpoint.